### DISCUSSION TITLE
can i set img_folder separately?

### DISCUSSION THREAD

**Discussion by Jaehaks** [epwalsh/obsidian.nvim#657] [Q&A] [answered]
i hope to make separated directory depends on note. 
can i set function as img_folder?

**Comment by superzanti** [comment]
If you are asking about having the img_folder be relative to the current note, I would very much like to do this. Anyone know of an easy way?

**Comment by epwalsh** [comment, marked as answer]
Yes to both. You use a note-dependent directory by overriding the `image_name_func` option:

```lua
      image_name_func = function()
        ---@type obsidian.Client
        local client = require("obsidian").get_client()

        local note = client:current_note()
        if note then
          return string.format("%s/", note.id)
        else
          return string.format("%s-", os.time())
        end
      end,
```

And if you want the folder to be relative to the current note, you could have `image_name_func` return an absolute path like this:

```lua
      image_name_func = function()
        ---@type obsidian.Client
        local client = require("obsidian").get_client()

        local note = client:current_note()
        if note then
          return string.format("%s/imgs/", note.path)
        else
          return string.format("%s-", os.time())
        end
      end,
```

In that case you may also want to update `img_text_func` to use a relative path in the final text that gets inserted:

```lua
  attachments = {
    -- A function that determines the text to insert in the note when pasting an image.
    -- It takes two arguments, the `obsidian.Client` and an `obsidian.Path` to the image file.
    -- This is the default implementation.
    ---@param client obsidian.Client
    ---@param path obsidian.Path the absolute path to the image file
    ---@return string
    img_text_func = function(client, path)
      path = client:vault_relative_path(path) or path
      return string.format("![%s](%s)", path.name, path)
    end,
  },
```

**Reply by Jaehaks** [reply]
Thank you for your exact solution :)

**Reply by Jaehaks** [reply]
@epwalsh 

I tried it, But it doesn't works unfortunately,
I think `img_name_func` doesn't override `img_folder`. 
Images always paste in assets/imgs declared by `img_folder`. 

If I enter filename with path to confirming message like 'test/test.png', 
it downloads the image to the entered path exactly,

**Reply by epwalsh** [reply]
@Jaehaks I think it would work if you passed an absolute path

**Reply by Jaehaks** [reply]
@epwalsh 
I did it also. i enter the absolute directory to img_name_func's retrun value. but it cannot recognize directory. 
is it because i am using windows directory?

**Reply by epwalsh** [reply]
@Jaehaks that's possible. Can you show me exactly what you're trying?

**Reply by Jaehaks** [reply]
@epwalsh 

OK, I set like this, 
```lua		
config = function ()
  local obsidian = require('obsidian')
  obsidian.setup({
    workspaces = {
      {
        name = 'personal',
        path = vim.fn.expand('$HOME') .. '\\Obsidian_Nvim\\personal'
      },
      {
        name = 'works',
        path = vim.fn.expand('$HOME') .. '\\Obsidian_Nvim\\works'
      }
    },
    mappings = {}, -- disable default keymapping
    preferred_link_style = 'markdown',
    attachments = {
      img_name_func = function () -- it doesn't work
      local client = obsidian.get_client()
      
      local note = client:current_note() 
      return 'c:\\Users\\USER\\img'
      -- if note then
      -- 	-- return string.format("%s/imgs/", note.path)
      -- 	return string.format("%s/imgs/", note.path.filename)
      -- else
      -- 	return string.format("%s-", os.time())
      -- end
      end
    }
  })
  
end
```

But the confirm message show default img_folder always.
![image](https://github.com/user-attachments/assets/f3c18740-6a08-4b7a-b501-816dc26a77c8)

Do I set wrong configuration?
